server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -386,22 +389,42 @@ public void onFailure(Throwable t)
       throw Throwables.propagate(e);
     }
 
-    Sequence<T> retVal = new BaseSequence<>(
-        new BaseSequence.IteratorMaker<T, JsonParserIterator<T>>()
-        {
-          @Override
-          public JsonParserIterator<T> make()
+    Sequence<T> retVal;
+    if (useV3QueryUrl) {
```

Comment:
I propose using a factory here with the two default implementations being a queryv2 and a queryv3 factory which can generate the proper return value.